JavaAssignmentZemoso/Assignment1/FilesInDirectory.java

```diff
@@ -0,0 +1,93 @@
+/*
+@author:- Chiranjiv Goyal
+
+Create a java program to search through the home directory and 
+look for files that match a regular expression. 
+The program should be able to take inputs repeatedly and 
+should print out the full absolute path of the matching files found.
+
+
+*/
+package Assignment1;
+
+import java.io.BufferedReader;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.io.IOException;
+import java.io.File;
+import java.io.InputStreamReader;
+
+public class FilesInDirectory
+{
+	/**
+	 * Print all files matching to regex in the home directory with there path
+	 * @throws IOException
+	 */
+
+	public void matchFilesInDirectory() throws IOException
+	{
+		/**
+			Call user for pattern to be given as input. Inserting 'N' will exit the program.
+		*/
+		String pattern = "";
+		BufferedReader reader =  new BufferedReader(new InputStreamReader(System.in));
+		
+		while(pattern != "N")
```

Comment:
How will this program handle the cases where I want to search all the files containing the character N?